src/connection/connection.go

```diff
@@ -0,0 +1,180 @@
+package connection
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/mock"
+)
+
+// Client is an interface for mocking
+type Client interface {
+	Brokers() []Broker
+	Topics() ([]string, error)
+	Partitions(string) ([]int32, error)
+	RefreshCoordinator(string) error
+	Coordinator(string) (Broker, error)
+	Leader(string, int32) (Broker, error)
+	Close() error
+	GetOffset(string, int32, int64) (int64, error)
+}
+
+// SaramaClient is a wrapper struct for sarama.Client
+type SaramaClient struct {
+	Inner sarama.Client
```

Comment:
Can we try embedding and just have the `Brokers` method? Can remove some of the overhead.